tmk_core/common/keyboard.c

```diff
@@ -64,20 +64,41 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
 #ifdef MATRIX_HAS_GHOST
+static uint16_t matrix_ghost_check[MATRIX_ROWS];
 static bool has_ghost_in_row(uint8_t row)
 {
-    matrix_row_t matrix_row = matrix_get_row(row);
-    // No ghost exists when less than 2 keys are down on the row
-    if (((matrix_row - 1) & matrix_row) == 0)
+    matrix_row_t matrix_row = (matrix_get_row(row) & matrix_ghost_check[row]);
+    /* No ghost exists when less than 2 keys are down on the row.
+    If there are "active" blanks in the matrix, the key can't be pressed by the user,
+    there is no doubt as to which keys are really being pressed.
+    The ghosts will be ignored, they are KC_NO.   */
+    if (((matrix_row - 1) & matrix_row) == 0){
         return false;
-
-    // Ghost occurs when the row shares column line with other row
+    }
+    // Ghost occurs when the row shares column line with other row, blanks in the matrix don't matter
+    // If there are more than two real keys pressed and they match another row's real keys, the row will be ignored.
     for (uint8_t i=0; i < MATRIX_ROWS; i++) {
-        if (i != row && (matrix_get_row(i) & matrix_row))
+        if (i != row && __builtin_popcount((matrix_get_row(i) & matrix_ghost_check[i]) & matrix_row) > 1){
```

Comment:
This code looks like it behaves differently than before, say we have the tiny keyboard with a 2x2 matrix, and we set the keys as the following 
```
10
10
```
And check for `has_ghost_in_row(0)`

So `matrix_row` is `01`, and for `i==1`, we have `if (1 != 0 && (10 & 10)` which previously returned true.

But with the new code, and all keys enabled we have `__builtin_popcount((10 & 11) & 10)` which is `__builtin_popcount(10 & 10)` = `builtin_popcount(10)` which is the same as 1, so the if statement evaluates to `if (1 != 0 && 1 > 1)` which is false. Therefore I believe that this pull request doesn't work as it should.